source/main/modules/window-manager/index.ts

```diff
@@ -227,6 +237,13 @@ export default class WindowManager {
       return
     }
 
+    this._mainWindow.on('show', () => {
+      const leaveAppRunning = Boolean(global.config.get('system.leaveAppRunning'))
+      if (leaveAppRunning) {
+        global.tray.add(() => this.showAnyWindow(), () => app.quit())
```

Comment:
Mh. This might cause trouble if the user opens the main window several times (since the tray would be added on and on and on).

Rather, what would be cool, is if the TrayProvider itself had the possibility of showing and hiding the Try depending on the Preference setting. That is: The tray Icon immediately shows/hides when the user clicks the checkbox. I think otherwise we might run into logical errors. (Plus we could remove that small piece of logic from here, so no one needed to query the TrayProvider ever, would make it more concise, wouldn't it?)

Reply:
> Rather, what would be cool, is if the TrayProvider itself had the possibility of showing and hiding the Try depending on the Preference setting. That is: The tray Icon immediately shows/hides when the user clicks the checkbox.

Immediately show/hide the tray is implemented in [L183 of index.ts](https://github.com/Zettlr/Zettlr/pull/2026/files#diff-87c7699f8e2c0214b588787d0bb57543927f3f8857444550bf0dbb35d5294c06R183).

We will investigate and fix the logic error and moving code to TrayProvider.

Reply:
Ah, yes: This line should be in the TrayProvider. So that the Tray Provider itself listens to the config changes and shows/hides the Tray on its own. This would keep the window manager a little bit more lean. And, we have a lot of space in the TrayProvider either way, so this makes sense. Also, it encapsulates the functionality a little bit more in one place!